### PR TITLE
fix: align mf2 number function to spec

### PR DESCRIPTION
**Pull request by lucacasonato**
https://github.com/tc39/proposal-intl-messageformat/pull/66

### CODE REVIEW

File: packages/mf2-messageformat/src/functions/number.ts [hunk 1]
```diff
@@ -61,23 +61,18 @@ export function number(
     Intl.PluralRulesOptions & { select?: 'exact' | 'cardinal' | 'ordinal' } = {
     localeMatcher
   };
-  switch (typeof input) {
-    case 'bigint':
-    case 'number':
-      value = input;
```
Comment (eemeli) [inline packages/mf2-messageformat/src/functions/number.ts:67]: This is left out in the change, and is necessary for numeric inputs.

**Review by eemeli — CHANGES_REQUESTED** [review]

File: packages/mf2-messageformat/src/functions/number.ts (hunk 1 above)
Comment (lucacasonato) [reply packages/mf2-messageformat/src/functions/number.ts:67]: Oh I see - I didn't realize value and input were separate variables. They are not separate in the spec. Maybe I could align them here?

Comment (eemeli) [reply packages/mf2-messageformat/src/functions/number.ts:67]: Aligning them would be fine, if you can get TS to be satisfied with that...

Comment (eemeli) [reply packages/mf2-messageformat/src/functions/number.ts:67]: Actually, it might've been not wanting to change an argument value that drove this design, and so changing line 59 to something like
```ts
let value = input;
```
would probably be fine.

Comment (lucacasonato) [reply packages/mf2-messageformat/src/functions/number.ts:67]: OK will do.

File: packages/mf2-fluent/src/fluent.test.ts
Comment (eemeli) [inline packages/mf2-fluent/src/fluent.test.ts:1]: Why the changes here? Not objecting, they just seem irrelevant?

**Review by eemeli — APPROVED** [review]

File: packages/mf2-fluent/src/fluent.test.ts
Comment (lucacasonato) [reply packages/mf2-fluent/src/fluent.test.ts:1]: The test did not pass on my `en-DK` system - the test was assuming the system locale is set to `en-US`.
